src/SwitcherProvider.js

```diff
@@ -26,15 +23,33 @@ export default function SwitcherProvider(props) {
     };
   }, []);
 
+  const providedMethods = useMemo(() => {
```

Comment:
why not `useCallback` here?

Reply:
that was my initial implementation actually. but it seemed better to memo the object passed into the provider so it wouldn't cause the consumers to rerender if the provider rerendered. so i just stuck everything in one memo to make it simpler.